submission/programs/adae.R

```diff
@@ -0,0 +1,350 @@
+  )  %>% 
+
+  # --------------------------------------------------------------------#
+  # CQ01NAM - Customized Query 01 Name                                  #
+  # --------------------------------------------------------------------#
+  # If AEDECOD contains any of the character strings of                 #
+  # ('APPLICATION', 'DERMATITIS', 'ERYTHEMA', 'BLISTER')                #
+  # OR if AEBODSYS='SKIN AND SUBC UTANEOUS TISSUE                       #
+  # DISORDERS' but AEDECOD is not in ('COLD SWEAT',                     #
+  #                                   'HYPERHIDROSIS', 'ALOPECIA') then #
+  # CQ01NAM='DERMATOLOGIC EVENTS' Otherwise  CQ01NAM=NULL               #
+  # --------------------------------------------------------------------#  
+  mutate ( CQ01NAM  = ifelse(str_detect(.$AEDECOD , "APPLICATION" ) |
+                             str_detect(.$AEDECOD , "DERMATITIS")   |
+                             str_detect(.$AEDECOD , "ERYTHEMA")     |
+                             str_detect(.$AEDECOD , "BLISTER")      | 
+                             str_detect(.$AEBODSYS , "SKIN AND SUBCUTANEOUS TISSUE DISORDERS") &
+                             !str_detect(.$AEDECOD , "COLD SWEAT|HYPERHIDROSIS|ALOPECIA")  , 
+                             "DERMATOLOGIC EVENTS" , 
+                             NA_character_) 
+  ) %>%
+
+  # --------------------------------------------------------------------#
+  # AOCC01FL - 1st Occurrence 01 Flag for CQ01                          #
+  # --------------------------------------------------------------------#
+  # Subset to CQ01NAM='' and TRTEMFL='Y' and sort by                    #
+  # Subject (USUBJID), Start Date (ASTDT), and Sequence                 #
+  # Number (AESEQ) and flag the first record (set                       #
+  # AOCC01FL=’Y’) within each Subject (Flag First Treatment             #
+  # Emergent Dermatological Event for Time to Event Analysis)           #
+  # --------------------------------------------------------------------#  
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID   ),
+      order = vars(  ASTDT, AESEQ),
+      new_var = AOCC01FL ,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & CQ01NAM =='DERMATOLOGIC EVENTS'
+  )  
+
+# --------------------------------------------- #
+# Check variables against define &              #
+# Assign dataset labels, var labels and formats #
+# --------------------------------------------- #
+adae <- adae0 %>% 
+  drop_unspec_vars(adae_spec) %>% # Check all variables specified are present and no more
+  check_ct_data(adae_spec, na_acceptable = TRUE) %>% # Checks all variables with CT only contain values within the CT
+  order_cols(adae_spec) %>% # Orders the columns according to the spec
+  sort_by_key(adae_spec) %>% 
+  xportr_df_label(adae_spec) %>% #dataset label 
+  xportr_label(adae_spec) #variable labels
+
+adae1 <- convert_blanks_to_na(adae) #blanks to NA
```

Comment:
That is not needed.

Reply:
removed.

Reply:
Please review and resolve to ensure you do not see it on your end.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.